packages/manager/src/features/PlacementGroups/PlacementGroupsDeleteModal.tsx

```diff
@@ -0,0 +1,150 @@
+  const {
+    error: unassignLinodeError,
+    mutateAsync: unassignLinodes,
+    reset: resetUnassignLinodes,
+  } = useUnassignLinodesFromPlacementGroup(selectedPlacementGroup?.id ?? -1);
+
+  const error = deletePlacementError || unassignLinodeError;
+
+  React.useEffect(() => {
+    if (open) {
+      resetDeletePlacementGroup();
+      resetUnassignLinodes();
+    }
+  }, [open, resetUnassignLinodes, resetDeletePlacementGroup]);
+
+  const handleUnassignLinode = async (linode: Linode) => {
+    const payload: UnassignLinodesFromPlacementGroupPayload = {
+      linodes: [linode.id],
+    };
+
+    await unassignLinodes(payload);
+  };
+
+  const onDelete = async () => {
+    await deletePlacementGroup();
+    onClose();
+  };
+
+  const placementGroupLabel = selectedPlacementGroup
+    ? `Placement Group ${selectedPlacementGroup?.label} (${
+        AFFINITY_TYPES[selectedPlacementGroup.affinity_type]
+      })`
+    : 'Placement Group';
+
+  const isDisabled = !selectedPlacementGroup || assignedLinodesCount > 0;
+
+  return (
+    <TypeToConfirmDialog
+      entity={{
+        action: 'deletion',
+        name: selectedPlacementGroup?.label,
+        primaryBtnText: 'Delete',
+        type: 'Placement Group',
+      }}
+      errors={
+        !selectedPlacementGroup
+          ? [{ reason: 'Placement Group not found.' }]
+          : undefined
+      }
+      inputProps={{
+        disabled: isDisabled,
+      }}
+      disabled={isDisabled}
+      label="Placement Group"
+      loading={placementGroupDataLoading || deletePlacementLoading}
+      onClick={onDelete}
+      onClose={onClose}
+      open={open}
+      title={`Delete ${placementGroupLabel}`}
+    >
+      {error && (
+        <Notice
+          key={selectedPlacementGroup?.id}
+          text={error?.[0]?.reason}
+          variant="error"
+        />
+      )}
+
+      <Notice spacingTop={8} variant="warning">
+        <Typography>
+          <strong>Warning:</strong>
+        </Typography>
+        <List
+          sx={(theme) => ({
+            '& > li': {
+              display: 'list-item',
+              fontSize: '0.875rem',
+              pb: 0,
+              pl: 0,
+            },
+            listStyle: 'disc',
+            ml: theme.spacing(2),
+            mt: theme.spacing(),
+          })}
+        >
+          <ListItem>
+            deleting a placement group is permanent and cannot be undone.
+          </ListItem>
+          <ListItem>
+            You need to unassign all Linodes before deleting a placement group.
+          </ListItem>
+        </List>
+      </Notice>
+      <RemovableSelectionsList
+        headerText={`Linodes assigned to ${placementGroupLabel}`}
+        id="assigned-linodes"
+        maxWidth={540}
+        noDataText="No Linodes assigned to this Placement Group."
+        onRemove={handleUnassignLinode}
+        removeButtonText="Unassign"
+        selectionData={assignedLinodes ?? []}
+        sx={{ mb: 3, mt: 1 }}
+      />
```

Comment:
Should we add a loading state to the unassign action?

Maybe we should allow a custom button to be passed to `RemovableSelectionsList` for more flexibility? Or button props at least

For example
```suggestion
      <RemovableSelectionsList
        headerText={`Linodes assigned to ${placementGroupLabel}`}
        id="assigned-linodes"
        maxWidth={540}
        noDataText="No Linodes assigned to this Placement Group."
        onRemove={handleUnassignLinode}
        removeButton={(props) => (
          <Button loading={isLoading}>Remove</Button>
        )}
        selectionData={assignedLinodes ?? []}
        sx={{ mb: 3, mt: 1 }}
      />
```

Reply:
I agree with this suggestion.

Reply:
Yup me too! done and loading state added